protocols/kad/src/behaviour.rs

```diff
@@ -753,6 +833,33 @@ where
                             self.queued_events.push(event);
                         }
                     },
+                    QueryInfoInner::GetValue { key: _, results } => {
+                        let result = match results.first() {
+                            Some(a) => Some(a.clone()),
+                            None => None,
+                        };
+
+                        let event = KademliaOut::GetValueRes {
+                            result: result.clone(),
+                            closer_peers: closer_peers.collect(),
+                        };
+
+                        break Async::Ready(NetworkBehaviourAction::GenerateEvent(event));
+                    },
+                    QueryInfoInner::PutValue { key, value } => {
+                        self.records.insert(key.clone(), value.clone());
```

Comment:
See the other comment about this seemingly duplicate insertion.

Reply:
Here we put the data on the side of the peer that received `PutValue`, as stated in the `TODO` the first one is probably not necessary

Reply:
Are you sure about that? As far as I'm aware only the sender creates a query and drives it to completion, so this code still runs on the side of the sender just like [this earlier insertion](https://github.com/libp2p/rust-libp2p/pull/1144/files#diff-88d683446d7fa7abbd7a347fb7d7c540R326), whereas the receiver adds it [here](https://github.com/libp2p/rust-libp2p/pull/1144/files#diff-88d683446d7fa7abbd7a347fb7d7c540R703). Am I really mistaken?

Reply:
Oh, yes you are right